### PR TITLE
LibGfx+LibWeb: Add support for canvas2d lineCap type "butt"

### PR DESCRIPTION
**Pull request by nico**
I mostly want to have support for this in LibGfx for LibPDF, but it's easier to write tests in html than in PDF.

The LibGfx code will also be useful for SVG, in addition to LibPDF.

The butt line ending implementation is kind of janky (see description of that commit), but hey, we can always improve it. It's less janky than the round line ending implementation :P

Before:

Screenshot 2024-09-24 at 10 22 15 PM

After:

Screenshot 2024-09-24 at 10 20 20 PM

<details>
<summary>source</summary>

```
<!DOCTYPE html>
<html>
<head>
<title>Line Cap Example</title>
<style>
  canvas { border: 1px solid #ccc; }
</style>
</head>
<body>
<canvas id="myCanvas" width="300" height="150"></canvas>
<script>
const canvas = document.getElementById('myCanvas');
const ctx = canvas.getContext('2d');

// Line styles
ctx.lineWidth = 10;

// Butt cap (default)
ctx.beginPath();
ctx.lineCap = 'butt';
ctx.moveTo(20, 30);
ctx.lineTo(100, 30);
ctx.stroke();

// Round cap
ctx.beginPath();
ctx.lineCap = 'round';
ctx.moveTo(20, 75);
ctx.lineTo(100, 75);
ctx.stroke();

// Square cap
ctx.beginPath();
ctx.lineCap = 'square';
ctx.moveTo(20, 120);
ctx.lineTo(100, 120);
ctx.stroke();

// Labeling
ctx.font = '12px Arial';
ctx.fillText('butt', 110, 30);
ctx.fillText('round', 110, 75);
ctx.fillText('square', 110, 120);
</script>
</body>
</html>
```
</details>

### CODE REVIEW

**Comment by nico** [comment]
@MacDue for path hacks; uh, @AtkinsSJ for libweb stuff maybe?

**Comment by nico** [comment]
Now with a dedicated test.

Before:

Screenshot 2024-09-25 at 8 43 19 AM

After:

Screenshot 2024-09-25 at 8 40 26 AM

Note that the second line now has straight(ish) ends, compared to round ends before. The third line should have a square ending, but that isn't implemented yet. The triangle shows caps both in closed and open form. It only should do that for the open path. That's an existing issue that just becomes visible with this change.

Here's how it looks in chrome:

Screenshot 2024-09-25 at 8 49 48 AM

File: Userland/Libraries/LibGfx/AntiAliasingPainter.cpp [hunk 1]
```diff
@@ -200,12 +200,12 @@ void AntiAliasingPainter::stroke_path(Path const& path, Color color, float thick
     fill_path(path.stroke_to_fill(thickness), color);
 }
 
-void AntiAliasingPainter::stroke_path(Path const& path, Gfx::PaintStyle const& paint_style, float thickness, float opacity)
+void AntiAliasingPainter::stroke_path(Path const& path, Gfx::PaintStyle const& paint_style, float thickness, float opacity, Path::CapStyle cap_style)
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/AntiAliasingPainter.cpp:203]: s/throgh/through in the commit message

**Review by LucasChollet — COMMENTED** [review]
Small fly-by comment

File: Userland/Libraries/LibGfx/AntiAliasingPainter.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/AntiAliasingPainter.cpp:203]: Done, thanks!

**Comment by nico** [comment]
Looks like the existing round cap code has inconsistent results on linux and mac. I'll add the test in a preparatory commit once I figured out how to make the round caps work consistently on macOS.

**Comment by nico** [comment]
#25042 is that prep.

**Pull request merged**